### PR TITLE
Fix bug in notifications

### PR DESCRIPTION
**Pull request by stsewd**
If an exception is raised before setting the self.build and self.version
objects, we ended up with those having the value of `{}` (empty dict).

In send_notification we expect to have these objects initialized.
We should refactor this to only have one point where to catch exceptions
and send the notification from there. Related to #5672

### CODE REVIEW

**Comment by stsewd** [comment]
Here is where these objects don't get initialized if the api call fails

https://github.com/rtfd/readthedocs.org/blob/2e797fa1b45fe9772c6cf77c4fd55731d515585d/readthedocs/projects/tasks.py#L381-L382

**Comment by stsewd** [comment]
No changes are needed for the coporate site

File: readthedocs/projects/tasks.py [hunk 1]
```diff
@@ -587,7 +587,7 @@ def run_build(self, docker, record):
                 log.warning('No build ID, not syncing files')
 
         if self.build_env.failed:
-            self.send_notifications()
+            self.send_notifications(self.version.pk, self.build['id'])
```
Comment (humitos) [inline readthedocs/projects/tasks.py]: Some lines above we are using `self.build.get('id')` and then `if build_id`. So, I suppose that we can't always access the `id` like here and we should probably use `build_id` directly which could be `None`.

**Comment by stsewd** [comment]
So, current flow granitize that if we ended up in the `run_build` function, no exception was raised while fetching the build object from the api (it's called from the else statement of the try block). Means, we always have an id.

I'm keeping the if statement just in case we mutate the object in another part of the code (not likely or I hope so p:). If the object ended up without an id or None we will see it in sentry. Current implementation has the same problem anyway.

File: readthedocs/projects/tasks.py [hunk 2]
```diff
@@ -566,15 +566,14 @@ def run_build(self, docker, record):
...
-                build_id = self.build.get('id')
             except vcs_support_utils.LockTimeout as e:
                 self.task.retry(exc=e, throw=False)
                 raise VersionLockedError
             except SoftTimeLimitExceeded:
                 raise BuildTimeoutError
 
             # Finalize build and update web servers
-            if build_id:
+            if self.build.get('id'):
```
Comment (davidfischer) [inline readthedocs/projects/tasks.py]: I had a slightly different fix for this in #5549.

**Pull request merged**